File: memory_test/logn/logn.java
import java.io.File;
import java.io.FileNotFoundException;
import java.math.BigDecimal;
import java.math.RoundingMode;
import java.util.Scanner;

public class logn {
    // Returns index of x if it is present in arr[].
    int binarySearch(int arr[], int x) {
        int low = 0, high = arr.length - 1;
        while (low <= high) {
            int mid = low + (high - low) / 2;

            // Check if x is present at mid
            if (arr[mid] == x)
                return mid;

            // If x greater, ignore left half
            if (arr[mid] < x)
                low = mid + 1;

            // If x is smaller, ignore right half
            else
                high = mid - 1;
        }

        // If we reach here, then element was
        // not present
        return -1;
    }

    // Driver code
    public static void main(String args[]) throws FileNotFoundException {
        Scanner reader = new Scanner(new File("/home/teaguy21/Documents/memory_test/constant/constant.txt"));
        int myNumber = reader.nextInt();
        reader.close();
        logn ob = new logn();
        int amount = myNumber;
        int[] inputArray = new int[amount];
        for (int i = 0; i < amount; i++) {
            inputArray[i] = i;
        }

        long start = System.nanoTime();
        ob.binarySearch(inputArray, 1);
        long end = System.nanoTime();
        BigDecimal d = new BigDecimal((end - start) / 1000000000.0).setScale(6, RoundingMode.HALF_EVEN);
        System.out.println(d);
        // if (result == -1)
        // System.out.println(
        // "Element is not present in array");
        // else
        // System.out.println("Element is present at "
        // + "index " + result);
    }
}

File: memory_test/logn/logn.js
function binarySearch(arr, x)
{    
    let low = 0;
    let high = arr.length - 1;
    let mid;
    while (high >= low) {
         mid = low + Math.floor((high - low) / 2);
 
        // If the element is present at the middle
        // itself
        if (arr[mid] == x)
            return mid;
 
        // If element is smaller than mid, then
        // it can only be present in left subarray
        if (arr[mid] > x)
            high = mid - 1;
            
        // Else the element can only be present
        // in right subarray
        else 
            low = mid + 1;
    }
 
    // We reach here when element is not
    // present in array
    return -1;
}

const myModule = require('/home/teaguy21/Documents/memory_test/constant/constant.js');

const arr = [];
for (let i = 1; i <= myModule.myVariable; i++) {
  arr.push(i);
}

const {performance} = require('perf_hooks');
const x = 1;

const startTime = performance.now(); 
const result = binarySearch(arr, x);
const endTime = performance.now();

console.log(`${((endTime - startTime) / 1000).toFixed(6)}s`);

// if (result !== -1) {
//   console.log(`Element is present at index ${result}`);
// } else {
//   console.log("Element is not present in array");
// }
